hooks/gov/useProposals.ts

```diff
@@ -0,0 +1,53 @@
+import { useQuery } from "react-query";
+import {
+  ProposalHookParams,
+  ProposalHookReturn,
+} from "./interfaces/hookParams";
+import { getCantoApiData } from "@/config/api";
+import { Proposal } from "./interfaces/proposal";
+import { NEW_ERROR, errMsg } from "@/config/interfaces";
+
+export default function useProposals(
+  params: ProposalHookParams,
+  options?: {
+    refetchInterval?: number;
+  }
+): ProposalHookReturn {
+  ///
+  /// INTERNAL HOOKS
+  ///
+
+  // just need to fetch all proposals for this hook
+  const { data: proposals } = useQuery(
+    ["proposals", params.chainId],
+    async () => {
+      const { data: proposals, error } = await getCantoApiData<string>(
```

Comment:
we can use the main api now that it works with proposal data, title, and description